### DISCUSSION TITLE
Should I use Instrumentation to make a connection to the database?

### DISCUSSION THREAD

**Discussion by Tik1993** [vercel/next.js#71523] [App Router]
I want to connect to MongoDB using Mongoose in Next.js, but I’m thinking about where I should place my connectDB function to handle the database connection. I find that Instrumentation is a good feature since the register function is called once when a new Next.js server instance is initiated.

If not, should I place the connectDB function in the root layout file, or inside every function that needs to connect to the database?

**Comment by omermecitoglu** [comment]
Im running drizzle migrations in instrumentation without any problem
